### PR TITLE
arm64: dts: qcom: msm8916-samsung-a5-zt: Add initial device tree

### PR DESCRIPTION
**Pull request by wonderfulShrineMaidenOfParadise**
Samsung Galaxy A5 (SM-A500YZ) is a smartphone using the MSM8916 SoC released
in 2015.

The A5 variants are very similar, with some differences in accelerometer and
touch key. The common parts are shared in msm8916-samsung-a5u-eur.dts to reduce
duplication.

The difference of accelerometer between A5U/A5ZT is the way the sensor is
mounted on the mainboard - set the mount-matrix in the
device-specific device tree part to handle that difference.

On A5ZT the touch key is supplied by a single fixed
regulator (enabled via GPIO 60) that supplies both MCU and LED.

A5ZT doesn't have an NFC chip.

### CODE REVIEW

**Comment by wonderfulShrineMaidenOfParadise** [comment]
Actually this is copied from a5u-eur, with little changes about GPIO from 97 to 60.
I guess there is a better solution than adding a dts directly like this.

File: arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+#include "msm8916-samsung-a5u-eur.dts"
+
+/ {
+	model = "Samsung Galaxy A5 (SM-A500YZ)";
+	compatible = "samsung,a5-zt", "qcom,msm8916";
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts:8]:
```suggestion
	compatible = "samsung,a5-zt", "samsung,a5u-eur", "qcom,msm8916";
```

since it's based on that dt but "extends" it. We will rename a5u-eur to just a5 and it will make more sense later.

**Comment by wonderfulShrineMaidenOfParadise** [comment]
Pushed another one with this:
```c
i2c-nfc {
    status = "disable";
};
```

File: arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+/ {
+	model = "Samsung Galaxy A5 (SM-A500YZ)";
+	compatible = "samsung,a5-zt", "samsung,a5u-eur", "qcom,msm8916";
+	chassis-type = "handset";
+
+	/* NOTE: Despite the fact that instead of S3FWRN5, PN547 is defined in
+	 * downstream, a5-zt doesn't actually have an NFC chip.
+	 */
+	i2c-nfc {
+		status = "disable";
```
Comment (TravMurav) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts:16]:
```suggestion
		status = "disabled";
```

And maybe reword the comment without referring to the downstream, just state the fact that nfc is missing on the board.

Also I don't think I deserve a "Co-authored-by" tag on this commit (more to that - with a wrong email :)

**Review by TravMurav — APPROVED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/dts-v1/;
+
+#include "msm8916-samsung-a5u-eur.dts"
+
+/ {
+	model = "Samsung Galaxy A5 (SM-A500YZ)";
+	compatible = "samsung,a5-zt", "samsung,a5u-eur", "qcom,msm8916";
+	chassis-type = "handset";
+
+	/* NOTE: a5-zt doesn't actually have an NFC chip. */
```
Comment (stephan-gh) [inline arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts:12]: Source? Did you check on the board or just functionally?

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts:12]: In other words, is there a chance that you just didn't set it up correctly in the device tree?

Comment (stephan-gh) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts:12]: Apparently clearly missing on the board, @TravMurav answered this one.

**Review by stephan-gh — APPROVED** [review]

File: arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts (hunk 3 above)
Comment (wonderfulShrineMaidenOfParadise) [reply arch/arm64/boot/dts/qcom/msm8916-samsung-a5-zt.dts:12]: For some variants `PN547` is defined in downstream. However it's not documented on the user manuals, and there is no NFC chip seen on my board either.

[SM-A5000](https://downloadcenter.samsung.com/content/UM/201508/20150817114520200/SM-A5000_UM_Open_HongKong_Lollipop_Eng_Rev.1.0_150804.pdf#p7)
[SM-A500YZ](https://downloadcenter.samsung.com/content/UM/201508/20150817114828408/SM-A500YZ_UM_Open_Taiwan_Lollipop_Chi_Rev.1.0_150813.pdf#p7)
